clang/tools/driver/cc1_main.cpp

```diff
@@ -523,12 +521,16 @@ Optional<int> CompileJobCache::tryReplayCachedResult(CompilerInstance &Clang) {
     return 1;
 
   Expected<bool> ReplayedResult =
-      CacheBackend->tryReplayCachedResult(*ResultCacheKey);
+      Clang.getFrontendOpts().DisableCachedCompileJobReplay
```

Comment:
Should we capture this flag and then cleared to false in `initialize` (or in`canonicalizeForCaching` after https://github.com/apple/llvm-project/pull/5286 lands)?  It won't matter for the current compilation since you're not replaying, but you could allow using the artifacts from this compilation in a later compilation that does not set the flag, I think.

Reply:
Good idea, but seems a bit better to clear it from `clang::createCompileJobCacheKey`, WDYT?
That way I won't need to keep it as additional state of `CompileJobCache`.

Reply:
See https://github.com/apple/llvm-project/pull/5286 for how I'm thinking we should handle this by having an API that modifies the invocation in-place and extracts the settings we need. The bad thing about modifying settings only in `createCompileJobCacheKey` is that there is no way to tell the rest of the compiler actually ignores these settings.  Right now we do that for output paths, but I want to eventually get rid of that as well and make it actually modify the invocation so they're not available outside cc1_main.

Reply:
SGTM, I'll add in `initialize` within the context of this PR.

Reply:
See updated commit.